src/dmd/nogc.d

```diff
@@ -206,14 +217,15 @@ public:
 
     override void visit(CatAssignExp e)
     {
+        /* CatAssignExp will exist in `__traits(compiles, ...)` and in the `.e1` branch of a `__ctfe ? :` CondExp.
+         * The other branch will be `_d_arrayappendcTX(e1, 1), e1[$-1]=e2` which will generate the warning about
+         * GC usage. See visit(CallExp).
+         */
```

Comment:
Hmm, could we change `visit(ConExp)` instead to not visit `__ctfe` only code? I guess that right now we probably do not differentiate, but in the long run it should be possible to ignore everything in `__ctfe` only blocks for nogc/codegen. As you already have the patch for codegen, why not do the same for nogc checks?

And then add a Test + Changelog entry for this. Some people might be quite happy about this change: https://run.dlang.io/is/G6fv55 ;-)

Reply:
Sadly I cannot change so it doesn't visit this function because of two reasons.
1. nogc.d is using `dmd.apply.walkPostorder` which walks it in this order: `walk(econd)`, `walk(e1)`, `walk(e2)`, and lastly `visit(CondExp)`. https://github.com/dlang/dmd/blob/master/src/dmd/apply.d#L144. As this walker is used all over the codebase I don't think it is a good idea to introduce changes to it.

2. It still needs to set `err = true` to mark the appending as `non-@nogc` because the lowering code will not lower code that is in a `__traits(compiles, ...)`. Printing error here is not needed because the error will always be gagged for `SCOPE.compile`, and if errors were printed they would print the exact same message as the one in `visit(CallExp)`.

So this visit function will now just silently error.